### PR TITLE
FXIOS-708 ⁃ [iOS14 Widgets] : iOS14 Quick Action Widgets (Medium/Small) Versions

### PR DESCRIPTION
**Pull request by noorhashem**
Parent Branch code : #6890. Thank you @q2r5 for your contributions.

- [X] Hooked up go to copied link feature to the SwiftUI design by @q2r5 

- [X] Hook up Close private tabs feature functionality to the design

- [X] Redesigned the Medium and Small Widgets according to new designs.

- [X] Configuration for small widgets

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/FXIOS-708)

### CODE REVIEW

File: Client/Frontend/Browser/TabManager.swift [hunk 1]
```diff
@@ -194,7 +194,7 @@ class TabManager: NSObject {
         } else {
             _selectedIndex = -1
         }
-        assert(_selectedIndex > -1, "Tab expected to be in `tabs`")
+//        assert(_selectedIndex > -1, "Tab expected to be in `tabs`")
```
Comment (noorhashem) [inline Client/Frontend/Browser/TabManager.swift:197]: For some reason, this line didn't make the feature work as expected (to be researched)

**Comment by dnarcese** [comment]
Let's update this design based on Amy's new mocks https://app.abstract.com/share/6eb2f2ec-3975-455b-9770-9488d05d8b5c?mode=design&sha=836345d162d60091b41d31f17b75619fa55f8fd2

File: Client/Frontend/Browser/TabManager.swift (hunk 1 above)
Comment (nbhasin2) [reply Client/Frontend/Browser/TabManager.swift:197]: If not needed then we could just get rid of it @noorhashem

File: Client/Frontend/Strings.swift [hunk 2]
```diff
@@ -797,3 +797,15 @@ extension Strings {
 extension String {
     public static let FxAQRCode_Instructions = NSLocalizedString("fxa.qr-scanning-view.instructions", value: "Scan the QR code shown at firefox.com/pair", comment: "Instructions shown on qr code scanning view")
 }
+
+//Today Widget Strings - [New Search - Private Search]
+extension String {
```
Comment (nbhasin2) [inline Client/Frontend/Strings.swift:802]: I thought we added these strings? Didn't we?

Comment (noorhashem) [reply Client/Frontend/Strings.swift:802]: Amy made changes to the normal search button text, I'll make it in a separate PR

**Comment by noorhashem** [comment]
> Unfortunately the blur won't work. `UIViewRepresentable` is not allowed in widgets.

@q2r5  Do you have any resources confirming this? then we know for sure, because it's supposed to work normally with SwiftUI views as far as I know.

**Comment by q2r5** [comment]
> > Unfortunately the blur won't work. `UIViewRepresentable` is not allowed in widgets.
> 
> 
> 
> @q2r5  Do you have any resources confirming this? then we know for sure, because it's supposed to work normally with SwiftUI views as far as I know.

These two answers on the forums from framework engineers state that `UIViewRepresentable` is not supported.

https://developer.apple.com/forums/thread/652042?answerId=617362022#617362022

https://developer.apple.com/forums/thread/653471?answerId=619627022#619627022

File: Client/Application/NavigationRouter.swift [hunk 3]
```diff
@@ -89,7 +90,12 @@ enum NavigationPath {
         } else if urlString.starts(with: "\(scheme)://fxa-signin"), components.valueForQuery("signin") != nil {
             self = .fxa(params: FxALaunchParams(query: url.getQuery()))
         } else if urlString.starts(with: "\(scheme)://open-url") {
-            let url = components.valueForQuery("url")?.asURL
+            var url: URL?
+            if components.query?.contains("clipboard") ?? false {
+                url = UIPasteboard.general.url
+            } else {
+                url = components.valueForQuery("url")?.asURL
+            }
```
Comment (nbhasin2) [inline Client/Application/NavigationRouter.swift:98]: Debating if we should clean this up @noorhashem what do you think? 
I am good either ways. 
```suggestion
url = (components.query?.contains("clipboard") ?? false) ? UIPasteboard.general.url : components.valueForQuery("url")?.asURL
```

File: Client/Info.plist [hunk 4]
```diff
@@ -6,20 +6,8 @@
 	<string>$(ADJUST_APP_TOKEN)</string>
```
Comment (nbhasin2) [inline Client/Info.plist:6]: Lets try to clean this file @noorhashem

File: Client/Frontend/Strings.swift [hunk 5]
```diff
@@ -797,3 +797,14 @@ extension Strings {
...
+    public static let NewPrivateTabButtonLabel = NSLocalizedString("TodayWidget.PrivateTabButtonLabelV1", tableName: "Today", value: "Private Search", comment: "Open New Private Tab button label")
```
Comment (nbhasin2) [inline Client/Frontend/Strings.swift:803]: All strings are merged on main with some refactoring so we should find a good way to rebase this branch with main

File: WidgetKit/QuickLink.swift [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+            return "smallPrivateMask"
+        case .copiedLink:
+            return "copy_link_icon"
+        case .closePrivateTabs:
+            return "delete"
+        }
+    }
+
+    public var headlineTxt: String {
+        return String(String.newSearchButtonLabel.split(separator: " ")[2])
```
Comment (nbhasin2) [inline WidgetKit/QuickLink.swift:28]: @noorhashem whats the purpose of separating headline text?

File: WidgetKit/QuickLink.swift [hunk 7]
```diff
@@ -0,0 +1,95 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import SwiftUI
+import Shared
+
+enum QuickLink: Int {
```
Comment (nbhasin2) [inline WidgetKit/QuickLink.swift:8]: @noorhashem Could we add a structure comment to know what file does

File: WidgetKit/QuickLink.swift [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+    case closePrivateTabs
+
+    public var imageName: String {
+        switch self {
+        case .search:
+            return "faviconFox"
+        case .privateSearch:
+            return "smallPrivateMask"
+        case .copiedLink:
+            return "copy_link_icon"
```
Comment (nbhasin2) [inline WidgetKit/QuickLink.swift:21]: nit: can we rename the icon to `copyLinkIcon` so it matches others

File: WidgetKit/QuickLink.swift [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+        case .closePrivateTabs:
+            return "delete"
+        }
+    }
+
+    public var headlineTxt: String {
+        return String(String.newSearchButtonLabel.split(separator: " ")[2])
+    }
+
+    public var captionTxt: String {
```
Comment (nbhasin2) [inline WidgetKit/QuickLink.swift:31]: nit: rename captionTxt to `captionText`

File: WidgetKit/SearchQuickLinksMedium/SearchQuickLinks.swift [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+struct SearchQuickLinksWigdet: Widget {
+    private let kind: String = "Quick Actions - Medium"
+
+    public var body: some WidgetConfiguration {
+        StaticConfiguration(kind: kind, provider: Provider(), placeholder: SearchQuickLinksEntryView()) { entry in
+            SearchQuickLinksEntryView()
+        }
+        .supportedFamilies([.systemMedium])
+        .configurationDisplayName("Firefox - Quick Actions")
+        .description("Access to frequent actions directly on your home screen.")
```
Comment (nbhasin2) [inline WidgetKit/SearchQuickLinksMedium/SearchQuickLinks.swift:55]: @noorhashem Will this get replaced with what we have in strings file?

File: WidgetKit/ImageButtonWithLabel.swift [hunk 11]
```diff
@@ -0,0 +1,83 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import SwiftUI
+
+struct ImageButtonWithLabel: View {
```
Comment (nbhasin2) [inline WidgetKit/ImageButtonWithLabel.swift:7]: nit: Add a basic structure on how the widget will look like 
```
----------------
Label  | Image  |
                |
---------------

File: Client/Application/NavigationRouter.swift (hunk 3 above)
Comment (noorhashem) [reply Client/Application/NavigationRouter.swift]: @nbhasin2 this is redundant code actually, it's outdated. I'll update it.

File: WidgetKit/QuickLink.swift (hunk 6 above)
Comment (noorhashem) [reply WidgetKit/QuickLink.swift:28]: @nbhasin2  this is for the separate "Firefox" large string in new search and new private search buttons in the latest design updates.

File: WidgetKit/SearchQuickLinksMedium/SearchQuickLinks.swift (hunk 10 above)
Comment (noorhashem) [reply WidgetKit/SearchQuickLinksMedium/SearchQuickLinks.swift:55]: @nishant2718  yes

**Pull request merged**